### PR TITLE
Add failing test fixture for RemoveUselessParamTagRector

### PR DESCRIPTION
**Pull request by enumag**
# Failing Test for RemoveUselessParamTagRector

Based on https://getrector.org/demo/1ebfb5f9-a1f1-6412-b1e0-b75ba4616957

Test case for https://github.com/rectorphp/rector/issues/6628

### CODE REVIEW

File: rules-tests/DeadCode/Rector/ClassMethod/RemoveUselessParamTagRector/Fixture/x.php.inc [hunk 1]
```diff
@@ -0,0 +1,14 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Rector\Tests\DeadCode\Rector\ClassMethod\RemoveUselessParamTagRector\Fixture;
+
+class X {
+	/**
+	 * @param Closure(int): bool $filter
```
Comment (samsonasik) [inline rules-tests/DeadCode/Rector/ClassMethod/RemoveUselessParamTagRector/Fixture/x.php.inc]: it possibly need to have `\` prefix as inside namespace, or add use statement for `Closure`: 

```suggestion
	 * @param \Closure(int): bool $filter
```

File: rules-tests/DeadCode/Rector/ClassMethod/RemoveUselessParamTagRector/Fixture/x.php.inc [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+	 */
+	static function filter(Closure $filter): array {
```
Comment (enumag) [inline rules-tests/DeadCode/Rector/ClassMethod/RemoveUselessParamTagRector/Fixture/x.php.inc]:
```suggestion
	static function filter(\Closure $filter): array {
```

**Comment by enumag** [comment]
@samsonasik Nice catch. I didn't notice that the "Create a Test" button added a namespace without fixing this.

**Comment by samsonasik** [comment]
@enumag I cherry-picked your commit at PR https://github.com/rectorphp/rector-src/pull/661

**Comment by enumag** [comment]
@samsonasik Thanks! I noticed that you fixed it in a class that's only meant for param annotations. I believe that closures in return annotations deserve the same treatment so I added another failing test here: https://github.com/rectorphp/rector-src/pull/662

EDIT: I tried to fix it myself the same way as you did in #661. Hopefully it's correct.

**Pull request closed without merging**